python/lsst/meas/extensions/scarlet/io.py

```diff
@@ -0,0 +1,808 @@
+        """Use the scarlet models to set HeavyFootprints for modeled sources
+
+        Parameters
+        ----------
+        catalog : `lsst.afw.table.SourceCatalog`
+            The catalog missing heavy footprints for deblended sources.
+        band : `str`
+            The name of the band that the catalog data describes.
+        psfModel : `lsst.afw.detection.Psf`
+            The observed PSF model for the catalog.
+            If `psfModel` is `None` and `redistributeCoadd` is `None`
+            then the models are not convolved.
+        redistributeCoadd : `lsst.afw.image.Exposure`
+            The exposure that is the source for flux re-distribution.
+            If `resdistributeCoadd` is `None` then flux re-distibution is
+            not performed.
+            If `redistributeCoadd is not `None` and `psfModel` is `None`,
+            then the PSF of the `resdistributeCoadd` is used.
+        removeScarletData : `bool`
+            Whether or not to remove `ScarletBlendData` for each blend
+            in order to save memory.
+        updateFluxColumns : `bool`
+            Whether or not to update the `deblend_*` columns in the catalog.
+            This should only be true when the input catalog schema already
+            contains those columns.
+        """
+        # Iterate over the blends, since flux re-distribution must be done on
+        # all of the children with the same parent
+        parents = catalog[catalog["parent"] == 0]
+        # Get the index of the model for the given band
+        bandIndex = self.filters.index(band)
+
+        for parentRecord in parents:
+            parentId = parentRecord.getId()
+
+            try:
+                blendModel = self.blends[parentId]
+            except KeyError:
+                warnings.warn(f"Unable to load models found for parent source {parentId}")
+                continue
+            updateBlendRecords(
+                blendData=blendModel,
+                catalog=catalog,
+                modelPsf=self.psf,
+                observedPsf=psfModel,
+                redistributeCoadd=redistributeCoadd,
+                bandIndex=bandIndex,
+                parentFootprint=parentRecord.getFootprint(),
+                updateFluxColumns=updateFluxColumns,
+            )
+
+            # Save memory by removing the data for the blend
+            if removeScarletData:
+                del self.blends[parentId]
+
+
+def updateBlendRecords(blendData, catalog, modelPsf, observedPsf, redistributeCoadd, bandIndex,
+                       parentFootprint, updateFluxColumns):
+    """Create footprints and update band-dependent columns in the catalog
+
+    Parameters
+    ----------
+    blendData : `ScarletBlendData`
+        Persistable data for the entire blend.
+    catalog : `lsst.afw.table.SourceCatalog`
+        The catalog that is being updated.
+    modelPsf : `numpy.ndarray`
+        The 2D model of the model PSF.
```

Comment:
```suggestion
        The 2D model of the PSF.
```
Or is "model" meaning two different things here?